src/ui/component/dayPicker/dayPicker.tsx

```diff
@@ -0,0 +1,67 @@
+import { type FC, useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import type { ActiveModifiers } from 'react-day-picker'
+import { DayPicker as RDayPicker } from 'react-day-picker'
+import { enUS, fr, de } from 'date-fns/locale'
+import 'react-day-picker/dist/style.css'
+import './dayPicker.scss'
+
+const localeMap = new Map([
+  ['en', enUS],
+  ['fr', fr],
+  ['de', de]
+])
+
+type DayPickerProps = {
+  onSelect: (
+    day?: Date,
+    // @ts-expect-error built-in ract-day-picker handler signature has optional first parameter
+    selectedDay: Date,
+    activeModifiers: ActiveModifiers,
```

Comment:
I'm not a big fan coupling the contract with the `react-day-picker` library, it violates the _dependency inversion principle_ .

Reply:
Thanks for the feedback! Didn't think about this principle. Will find a way to address it.

Reply:
Simply addressed because of [Function Parameter Bivariance](https://www.typescriptlang.org/docs/handbook/type-compatibility.html#function-parameter-bivariance). Didn't know that TS allows it. Since we don't need other params, we can solve it this way.